lib/Sema/MiscDiagnostics.cpp

```diff
@@ -4748,19 +4748,20 @@ static void diagUnqualifiedAccessToMethodNamedSelf(const Expr *E,
           if (auto typeContext = DC->getInnermostTypeContext()) {
             // self() is not easily confusable
             if (!isa<CallExpr>(Parent.getAsExpr())) {
-
               auto baseType = typeContext->getDeclaredInterfaceType();
-              auto baseTypeString = baseType.getString();
+              if (baseType->getEnumOrBoundGenericEnum() == nullptr) {
```

Comment:
Nit: Can use ! instead of == nullptr here

Reply:
sg done